Wrappers/Python/test/test_DataContainer.py

```diff
@@ -99,261 +87,149 @@ def testGb_creation_nocopy(self):
-        #print("a refcount " , sys.getrefcount(a))
         ds = DataContainer(a, False, ['X', 'Y', 'Z'])
-        #ds.__iadd__( 2 )
         ds += 2
-        steps.append(timer())
         self.assertEqual(ds.as_array()[0][0][0], 3.)
-        #ds.__isub__( 2 )
         ds -= 2
-        steps.append(timer())
         self.assertEqual(ds.as_array()[0][0][0], 1.)
-        #ds.__imul__( 2 )
         ds *= 2
-        steps.append(timer())
         self.assertEqual(ds.as_array()[0][0][0], 2.)
-        #ds.__idiv__( 2 )
         ds /= 2
-        steps.append(timer())
         self.assertEqual(ds.as_array()[0][0][0], 1.)
 
         ds1 = ds.copy()
-        #ds1.__iadd__( 1 )
         ds1 += 1
-        #ds.__iadd__( ds1 )
         ds += ds1
-        steps.append(timer())
         self.assertEqual(ds.as_array()[0][0][0], 3.)
-        #ds.__isub__( ds1 )
         ds -= ds1
-        steps.append(timer())
         self.assertEqual(ds.as_array()[0][0][0], 1.)
-        #ds.__imul__( ds1 )
         ds *= ds1
-        steps.append(timer())
         self.assertEqual(ds.as_array()[0][0][0], 2.)
-        #ds.__idiv__( ds1 )
         ds /= ds1
-        steps.append(timer())
         self.assertEqual(ds.as_array()[0][0][0], 1.)
 
-
     def test_unary_operations(self):
-        X, Y, Z = 1024, 512, 512
         X, Y, Z = 256, 512, 512
-        steps = [timer()]
         a = -numpy.ones((X, Y, Z), dtype='float32')
-        steps.append(timer())
-        t0 = dt(steps)
-        #print("a refcount " , sys.getrefcount(a))
         ds = DataContainer(a, False, ['X', 'Y', 'Z'])
 
         ds.sign(out=ds)
-        steps.append(timer())
         self.assertEqual(ds.as_array()[0][0][0], -1.)
 
         ds.abs(out=ds)
-        steps.append(timer())
         self.assertEqual(ds.as_array()[0][0][0], 1.)
 
-        ds.__imul__(2)
+        ds *= 2
         ds.sqrt(out=ds)
-        steps.append(timer())
         self.assertEqual(ds.as_array()[0][0][0],
                          numpy.sqrt(2., dtype='float32'))
 
-    def test_binary_operations(self):
-        self.binary_add()
-        self.binary_subtract()
-        self.binary_multiply()
-        self.binary_divide()
-
-    def binary_add(self):
-        X, Y, Z = 512, 512, 512
-        #X, Y, Z = 1024, 512, 512
-        steps = [timer()]
+    def test_binary_add(self):
+        X, Y, Z = 128, 512, 512
         a = numpy.ones((X, Y, Z), dtype='float32')
-        steps.append(timer())
-        t0 = dt(steps)
 
-        #print("a refcount " , sys.getrefcount(a))
         ds = DataContainer(a, False, ['X', 'Y', 'Z'])
         ds1 = ds.copy()
         out = ds.copy()
 
-        steps.append(timer())
+        t0 = time()
         ds.add(ds1, out=out)
-        steps.append(timer())
-        t1 = dt(steps)
-        steps.append(timer())
+        t1 = time()
         ds2 = ds.add(ds1)
-        steps.append(timer())
-        t2 = dt(steps)
-
-        #self.assertLess(t1, t2)
+        t2 = time()
+        self.assertLess(t1 - t0, t2 - t1)
```

Comment:
If I remember correctly this test would only work if the data is very large.

Suggestion leave out the test of speed, and just keep the test of equality, and further reduce the size of the data being tested.